BWC tests should use portable path separators

**Symptom**

The `createClone` task in the `:distribution:bwc-zip` project creates its checkout directory with forward slashes: `String checkoutDir = "${buildDir}/bwc/checkout-5.x"`.

On Windows builds, we see the following faillure (see [example build failure](http://build-us-00.elastic.co/job/es_core_master_windows-2012-r2/5156/console)):

```
Executing task ':distribution:bwc-zip:createClone' (up-to-date check took 0.0 secs) due to:
Task has not declared any outputs.
Starting process 'command 'git''. Working directory: C:\jenkins\workspace\es_core_master_windows-2012-r2\distribution\bwc-zip Command: git clone C:\jenkins\workspace\es_core_master_windows-2012-r2 C:\jenkins\workspace\es_core_master_windows-2012-r2\distribution\bwc-zip\build/bwc/checkout-5.x
:distribution:bwc-zip:createClone FAILED
:distribution:bwc-zip:createClone (Thread[main,5,main]) completed. Took 0.005 secs.
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':distribution:bwc-zip:createClone'.
> A problem occurred starting process 'command 'git''
```

The checkout directory uses mixed path separators (i.e. `C:\jenkins\workspace\es_core_master_windows-2012-r2\distribution\bwc-zip\build/bwc/checkout-5.x`) which may be the cause of the problem.

**Expectation**

The checkout path is rendered on each platform using the native file separator, i.e. on Windows `C:\jenkins\workspace\es_core_master_windows-2012-r2\distribution\bwc-zip\build\bwc\checkout-5.x`.

--- a/distribution/bwc-zip/build.gradle
+++ b/distribution/bwc-zip/build.gradle
@@ -29,9 +29,9 @@ import org.elasticsearch.gradle.LoggedExec
 
 apply plugin: 'distribution'
 
-String checkoutDir = "${buildDir}/bwc/checkout-5.x"
+File checkoutDir = file("${buildDir}/bwc/checkout-5.x")
 task createClone(type: LoggedExec) {
-  onlyIf { new File(checkoutDir).exists() == false }
+  onlyIf { checkoutDir.exists() == false }
   commandLine = ['git', 'clone', rootDir, checkoutDir]
 }
 

